packages/compartment-mapper/src/parse-pre-cjs.js

```diff
@@ -0,0 +1,62 @@
+// @ts-check
+
+import { decodeSyrup } from '@endo/syrup/decode';
+
+const { freeze } = Object;
+
+/** @type {ParseFn} */
+export const parsePreCjs = async (
+  bytes,
+  _specifier,
+  location,
+  _packageLocation,
+) => {
+  const { source, imports, exports, reexports } = decodeSyrup(bytes, {
+    name: location,
+  });
+
+  /**
+   * @param {Object} moduleExports
+   * @param {Compartment} compartment
+   * @param {Record<string, string>} resolvedImports
+   */
+  const execute = async (moduleExports, compartment, resolvedImports) => {
+    const functor = compartment.evaluate(source);
+
+    const module = freeze({
+      get exports() {
+        return moduleExports;
+      },
+      set exports(value) {
+        moduleExports.default = value;
+      },
+    });
+
+    const require = freeze((/** @type {string} */ importSpecifier) => {
+      const namespace = compartment.importNow(resolvedImports[importSpecifier]);
```

Comment:
For use outside SES I'd expect the value of `compartment.importNow` to be corruptible. In fact, I would not be surprised if it is corruptible inside SES by putting an `importNow` on the compartment instance. I think all such internal calls should use an incorruptible internal `importNow` function, which is also what I'd expect to specify.